contrib/gitian-descriptors/make_git_archive

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# A helper script to be sourced into the gitian descriptors
+
+mkdir -p ${OUTDIR}/src
+RECENT_TAG=$(git describe --abbrev=0 HEAD)
+if [ $RECENT_TAG = $(git describe HEAD) ]; then
+  if [[ $RECENT_TAG == v* ]]; then
+    VERSION=${RECENT_TAG:1}
+  else
+    VERSION=$RECENT_TAG
```

Comment:
what is this condition for?

Reply:
We tag releases with the prepended "v", like "v0.20.0rc1".

This `if` statement just removes it on purpose to include the version string (e.g., "0.20.0rc1") into file names.

Reply:
Oh, I meant the else branch

Reply:
```
if [ $RECENT_TAG = $(git describe HEAD) ]; then
  if [[ $RECENT_TAG == v* ]]; then
    VERSION=${RECENT_TAG:1} # Set VERSION to release (with "v") tagged commit
  else
    VERSION=$RECENT_TAG # Set VERSION to non-release (without "v") tagged commit
  fi
else
  VERSION=$(git rev-parse --short HEAD) # Set VERSION to untagged commit
fi
DISTNAME=bitcoin-${VERSION} # Use VERSION in file names
```

Reply:
Why would there be a non-v* tagged commit? This should probably be an assert(false)?

Reply:
It is convenient for making gitian test builds. At least, for me :)

Reply:
ok, will leave this for other reviewers to bikeshed :)

Reply:
> It is convenient for making gitian test builds. At least, for me :)

I mentioned on IRC, however I'll just reiterate here, I'd definitely normally NACK unclear/undocumented additions to our build system / release scripts. However given that this should be removed in a follow up PR I think it's ok as is. If you've got patches that are helpful for testing, you can maintain them locally, but please don't just bundle them into a PR along with other changes.

Reply:
**The rationale for adding that addition:**
Currently, the `gitian-build.py` script allows to make builds from personal repos by passing the `--url` command line argument. Also a branch could be chosen. This change adds an option to point the script to a tagged commit on personal repo as well.